### PR TITLE
[Feature] Asset skills no longer required on assessment plan

### LINKED ISSUE #10044: Asset skills not required on assessment plan
## Details

**Decision: we are gonna keep it simple for now and just not require asset skills to be assessed on the plan.**

> During testing, every user raised the following complaint
> 
> > "We dont actually assess 'Asset' skill, why are you forcing me to add an assessment for it?"
> 
> This is a complicated question, since we will NOT be requiring them to assess it, but NOT planning for it will make it difficult to include it later on if necessary. Should we force users to plan for it? or should we create a feature that allows Assessment plans to be amended somehow?
> 
> We could still benefit from making it clear right now that "Adding a plan for your assets does not mean you need to perform those assessments" The 'instructions' should probably include something to that effect.

## Copy update
This small copy update. Insert "essential" before the word skill.

Image

### English
https://github.com/GCTC-NTGC/gc-digital-talent/blob/5c259f4a821bf4826a944f07ad849a721e35a0f7/apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx#L184-L190

### French
https://github.com/GCTC-NTGC/gc-digital-talent/blob/5c259f4a821bf4826a944f07ad849a721e35a0f7/apps/web/src/lang/fr.json#L378-L381

## Acceptance criteria
- [ ] Asset skills are not required to be assessed for the assessment plan to be complete.
- [ ] Asset skills do not show up on the list of "missing" skills.
- [ ] Copy updated to reflect that only essential skills are required.

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #10044

## 👋 Introduction

Make it so that an assessment plan no longer requires the assessing of asset skills.

## 🧪 Testing

1. Make a pool with essential and asset skills
2. Ensure your plan is complete and the pool can be published without attaching an assessment to some asset skills

Note, given technical skills sync to the application step, it will need to be a behavioural asset skill

## 📸 Screenshot

Copy update

image

Asset behavioural skill without assessments

image

The section crossed out does _not_ appear 

image

Plan complete

image

Can publish

image

### CODE REVIEW

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx [hunk 1]
```diff
@@ -45,8 +45,8 @@ const pageTitle = defineMessage(messages.assessmentPlan);
 
 const pageSubtitle = defineMessage({
   defaultMessage:
-    "Select, organize and define the assessments used to evaluate each skill in the advertisement. Make sure every skill is assessed at least once to complete your assessment plan.",
-  id: "SSZY5w",
+    "Select, organize and define the assessments used to evaluate each skill in the advertisement. Make sure every essential skill is assessed at least once to complete your assessment plan.",
+  id: "iuA2pt",
   description: "Subtitle for the assessment plan builder",
```
Comment (vd1992) [inline apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx:50]: Orphaned French string for this is 

```
  "SSZY5w": {
    "defaultMessage": "Sélectionner, organiser et définir les évaluations ayant été utilisées pour évaluer chaque compétence de l’offre d’emploi. 
Veillez à ce que chaque compétence soit évaluée au moins une fois pour compléter votre plan d’évaluation.",
    "description": "Subtitle for the assessment plan builder"
  },
```
Do I resend the whole object for translation or does someone know what the French change should be?

Comment (tristan-orourke) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx:50]: Here you go, use this:
`Sélectionner, organiser et définir les évaluations ayant été utilisées pour évaluer chaque compétence de l’offre d’emploi. Veillez à ce que chaque compétence essentielle soit évaluée au moins une fois pour compléter votre plan d’évaluation.`

Comment (vd1992) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx:50]: Thank you! 
https://github.com/GCTC-NTGC/gc-digital-talent/pull/10195/commits/cd9e0cedd0901745afc38eefd9f5cdac73e7559b

**Comment by brindasasi** [comment]
code changes looking good to me. Still trying to test but my local is giving me hard time. Refreshing everything I know .. hang on :(

**Comment by brindasasi** [comment]
Actual acceptance criteria works for a new process
1. Created a new process, 
2. created Essential & Asset Skills.
3. Left out Asset skill without assessment step. 
I had  `Complete` pill for the assessment section 
I had `Complete` pill for the process section 
and able to publish successfully. 


I tried samething with an existing process and repeated step 2 & 3. 
I had  `Complete` pill for the assessment section 
But I still get "Incomplete" pill for process section and couldn't publish the pool. 
Screenshot 2024-04-30 at 5 38 30 PM

I tried cloning the process and still getting the same issue. Couldn't publish that pool.
https://github.com/GCTC-NTGC/gc-digital-talent/assets/89482910/d6b12871-861b-4de5-aca3-42081c74918d

Probably diff issue based on the code changes. But would like to confirm it. Could you check it out?

**Comment by vd1992** [comment]
This is (should be) only touching the assessment plan status so pool completeness shouldn't be changed here. 
Pool completeness comes from `PoolIsCompleteValidator.php` and this could stem from an awkward seeding instance.

Though, I am curious as to what is triggering incomplete, whether it is something to be concerned about. 

image

**Comment by vd1992** [comment]
Publish the pool via API and see what happens, so from 
http://localhost:8000/graphiql

**Comment by brindasasi** [comment]
> Publish the pool via API and see what happens, so from http://localhost:8000/graphiql

Found it. Its a sep issue :) 
I had a location but only gave in English. Didn't specify in French. Below validator fails on that.
Frontend just kept quiet about that. It didn't prompt me it is required. 

Screenshot 2024-05-01 at 10 02 39 AM

**Review by brindasasi — APPROVED** [review]
Changes looks good to me 🥳. Will create a sep issue for pool complete validation.

**Pull request merged**